apps/server/src/modules/common-cartridge/common-cartridge-client/card-client/card-client.adapter.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { faker } from '@faker-js/faker';
+import { createMock, DeepMocked } from '@golevelup/ts-jest';
+import { REQUEST } from '@nestjs/core';
+import { Test, TestingModule } from '@nestjs/testing';
+import { AxiosResponse } from 'axios';
+import { Request } from 'express';
+import { UnauthorizedException } from '@nestjs/common';
+import { CardListResponse, CardResponse } from './cards-api-client/models';
+import { CardClientAdapter } from './card-client.adapter';
+import { BoardCardApi } from './cards-api-client';
+
+const jwtToken = faker.string.uuid();
+
+describe(CardClientAdapter.name, () => {
+	let module: TestingModule;
+	let adapterUnderTest: CardClientAdapter;
+	let cardApiMock: DeepMocked<BoardCardApi>;
+
+	beforeAll(async () => {
+		module = await Test.createTestingModule({
+			providers: [
+				CardClientAdapter,
+				{
+					provide: BoardCardApi,
+					useValue: createMock<BoardCardApi>(),
+				},
+				{
+					provide: REQUEST,
+					useValue: createMock<Request>({
+						headers: {
+							authorization: `Bearer ${jwtToken}`,
+						},
+					}),
+				},
+			],
+		}).compile();
+		adapterUnderTest = module.get(CardClientAdapter);
+		cardApiMock = module.get(BoardCardApi);
+	});
+
+	afterAll(async () => {
+		await module.close();
+	});
+
+	afterEach(() => {
+		jest.resetAllMocks();
+	});
+
+	it('CardClientAdapter should be defind', () => {
+		expect(adapterUnderTest).toBeDefined();
+	});
+
+	describe('getAllBoardCardsByIds', () => {
+		const setup = () => {
```

Comment:
a second describe statment containing the setup method is required here
```suggestion
		describe('when getAllBoardCardsByIds is called', () => {
            const setup = () => .....
        })
```